### PR TITLE
feat(interview): add new api & deprecation marks to the old

### PR DESCRIPTION
**Pull request by aaliakseyenka**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 💡 Background and solution

Adds new api & marks deprecated api

### CODE REVIEW

File: nestjs/src/courses/interviews/interviews.controller.ts [hunk 1]
```diff
@@ -96,4 +106,59 @@ export class InterviewsController {
...
     throw new BadRequestException('Invalid interview id');
   }
+
+  // use `type` as a way to differentiate between stage-interview and interview.
+  @Get('/:interviewId/:type/feedback')
+  @ApiOkResponse({ type: InterviewFeedbackDto })
+  @ApiForbiddenResponse()
+  @ApiBadRequestResponse()
+  @ApiOperation({ operationId: 'getInterviewFeedback' })
+  @RequiredRoles([CourseRole.Mentor, CourseRole.Supervisor, CourseRole.Manager])
```
Comment (valerydluski) [inline nestjs/src/courses/interviews/interviews.controller.ts:116]: @RequiredRoles([CourseRole.Mentor, CourseRole.Supervisor, CourseRole.Manager], true)

File: nestjs/src/courses/interviews/interviews.controller.ts [hunk 2]
```diff
@@ -96,4 +106,59 @@ export class InterviewsController {
...
+
+    return new InterviewFeedbackDto(interview);
+  }
+
+  @Post('/:interviewId/:type/feedback')
+  @ApiOkResponse()
+  @ApiForbiddenResponse()
+  @ApiBadRequestResponse()
+  @ApiOperation({ operationId: 'createInterviewFeedback' })
+  @RequiredRoles([CourseRole.Mentor, CourseRole.Supervisor, CourseRole.Manager])
```
Comment (valerydluski) [inline nestjs/src/courses/interviews/interviews.controller.ts:139]: @RequiredRoles([CourseRole.Mentor, CourseRole.Supervisor, CourseRole.Manager], true)

Comment (aaliakseyenka) [reply nestjs/src/courses/interviews/interviews.controller.ts]: you are my hero, thanks!

File: nestjs/src/courses/interviews/interviews.controller.ts (hunk 1 above)
Comment (aaliakseyenka) [reply nestjs/src/courses/interviews/interviews.controller.ts]: you are my double hero, thanks!

**Comment by valerydluski** [comment]
Approval with a recommendation to create an issue for the removal of deprecated methods in the server at some point

**Review by valerydluski — APPROVED** [review]

**Comment by aaliakseyenka** [comment]
https://github.com/rolling-scopes/rsschool-app/issues/2281

**Pull request merged**